mvc/views/razor.rst

```diff
@@ -1,8 +1,283 @@
+***************************
+One of my favorite features of razor is the ability to use application relative
+paths.
+
+.. code-block:: c#
+
+  <a href="~/Home">Home page</a>
+  <script type='text/javascript' src="~/some.js">
+
+This only works for attributes that are defined in HTML5 as urls. If you want
+to resolve a path in an attribute not defined as a url you can use
+
+.. code-block:: c#
+
+  <a cust-attr="@Url.Content("~/the-url")" href="#"> Link </a>
+
+***************
+Tag Attributes
+***************
+Razor makes it easy to have programmatic classes and ids for you html tags.
+
+.. code-block:: c#
+
+  @{
+    var AnchorClass = "dark link";
+    var AnchorId = "main-link";
+  }
+
+  <a href="#" class="@AnchorClass" id="@AnchorId"> Link </a>
+
+The output for this markup is below.
+
+.. code-block:: html
+
+  <a href="#" class="dark link" id="main-link"> Link </a>
+
+When the value you pass into an attribute is null razor will remove that
+attribute completly.
+
+.. code-block:: c#
+
+  @{
+    string classes = null;
+  }
+
+  <a href="#" class="@classes"> Link </a>
+
+.. code-block:: Html
+
+  <a href="#"> Link </a>
+
+If you want the attribute to exist, but to be empty then you use an empty string.
+
+.. code-block:: c#
+
+  @{
+    string classes = "";
+  }
+
+  <a href="#" class="@classes"> Link </a>
+
+.. code-block:: Html
+
+  <a href="#" class=""> Link </a>
+
+It doesn't just work with strings, but boolean values as well. This is useful
+for programatically disabling input fields, checking checkboxes, etc.
+
+.. code-block:: c#
+
+  @{
+    bool agree = true;
+    bool enabled = false;
+  }
+
+  <input type="checkbox" name="agree" value="agree" checked="@agree"/>
+  <input type="checkbox" name="option" value="option" disabled="@enabled"/>
+
+.. code-block:: html
+
+  <input type="checkbox" name="agree" value="agree" checked="checked"/>
+  <input type="checkbox" name="option" value="option" disabled="disabled"/>
+
+*********************
+Dependency Injection
+*********************
+Razor now supports dependency injection. View specific services can be injected
+independent of the controller.
+
+.. code-block:: c#
+
+  @inject IInjectedServiceInterface InjectedServiceName
+
+A more in depth guide can be found here :doc:`dependency-injection`
+
+******
+Async
+******
+Razor can render the results of an asynchronous method call through
+the use of ``@await Awaitable``
+
+.. code-block:: c#
+
+  <p>@await GetNameAsync()</p>
+  <p>@await Task.FromResult("Foo")</p>
+
+In addition, inside of the layout file you can flush the page. The logic behind
+this is so that while the main content of your page is being processed and built
+by razor, important sections containing css or your navbar can be sent out ahead
+early.
+
+.. code-block:: c#
+
+  @await FlushAsync()
+
+**********
+@inherits
+**********
+``@inherits`` Allows you to override the class that view inherits from. By
+default the view inherits from ``System.Web.Mvc.WebViewPage``.
+
```

Comment:
Should add a bit to this section and the inject section mentioning the capabilities of `@inherits MyBaseType<TModel>`. Aka we MVC injects the current model in place of `<TModel>` for inherits and inject.

/cc @pranavkm